lib/execution_engine2/sdk/EE2Runjob.py

```diff
@@ -579,15 +580,19 @@ def _retry(self, job_id: str, job: Job, parent_job: Job, as_admin: bool = False)
                     exception=e,
                 )
 
-        # 2) Notify the retry_parent that it has been retried
+        # 2) Notify the retry_parent that it has been retried by adding a retry id
         try:
-            job.modify(inc__retry_count=1)
+            job.modify(push__retry_ids=retry_job_id)
         except Exception as e:
             self._db_update_failure(
                 job_that_failed_operation=str(job.id),
                 job_to_abort=retry_job_id,
                 exception=e,
             )
+        # If the retry_ids is updated and if present, the child_jobs, is updated, set toggle to true
+        retry_job = self.sdkmr.get_mongo_util().get_job(job_id=retry_job_id)
+        retry_job.retry_saved_toggle = True
```

Comment:
It seems safer to set just this field rather than saving the entire job in case there's a state change from another part of the code between `get_job` and `save_job`. Especially if we thread retries - the jobs could be `run` and then the state change from `created` to `queued` could collide with this modification.

Reply:
good catch, changed

Reply:
:+1: